src/fondant/pipeline/pipeline.py

```diff
@@ -401,7 +387,16 @@ def read(
             cluster_type=cluster_type,
             client_kwargs=client_kwargs,
         )
-        return self._apply(operation)
+
+        manifest = Manifest.create(
+            pipeline_name=self.name,
+            base_path=self.base_path,
+            run_id=self.get_run_id(),
```

Comment:
This creates a new `run_id` every time it's called. I'm not sure yet how we should handle this:
- For static propagation, there is no run yet, so maybe it should not be required
- For eager execution, we might execute a component multiple times while iterating. How should these executions relate to the pipeline run id?